### DISCUSSION TITLE
Should we use xfrm drop policies for securing roardwarrior ?

### DISCUSSION THREAD

**Discussion by p088gll** [strongswan/strongswan#2187] [Q&A] [answered]
We are using charon-systemd to configure an ubuntu 22.04. IPSEC roadwarrior laptops  setup with dual-stack configuration. A Fortigate firewall is the remote endpoint.

We have the additional requirement that when the machine has no IPSEC connection/tunnel, it should be closed down completely except for DHCP, NTP and ICMP (pass MTU discovery!)  communication.  On the other hand, when connected to the friendly net, a set of already existing iptables rules should be in place. 

After reading about xfrm drop policies and shunting in the Strongswan documentation I thought that using a configuration with children in block and pass mode might do this. When the tunnels are down, the block&pass policies take precedence. When the tunnels are up, the tunnel SPs are used. In a friendly-net situation the strongswan charon daemon is simply stopped and iptables takes over. 

The swanctl configuration is included here 
```
connections {
  ba2 {
    dpd_delay = 10s
    remote_addrs = 1.2.3.4
    version = 2
    mobike = yes
    reauth_time = 10800
    proposals = aes256gcm16-prfsha384-curve25519 # DH 31
    vips = ::,0.0.0.0
    local {
      auth = ike:pubkey-sha256-sha384
      cert-yk {
        slot = 0
        handle = 1
        module = yubikey
      }
    }
    remote {
      auth = ike:pubkey-sha256-sha384
      id = "E=test@example.com, CN=*, OU=TestOrg, O=Test, C=DE"
      cacerts = ROOT_CA_2022.pem
    }
    children {
      ba2-v4 {
        dpd_action = trap
        remote_ts = 0.0.0.0/0
        esp_proposals = aes256gcm16-curve25519 # DH 31
        rekey_time = 3600
        start_action = start
        policies_fwd_out = yes
      }
      ba2-v6 {
        dpd_action = trap
        local_ts = 2a02:x:y:z::/64
        remote_ts = 2a02:x:y::/40
        esp_proposals = aes256gcm16-curve25519 # DH 31
        rekey_time = 3600
        start_action = start
        policies_fwd_out = yes
      }
    }
  }
  shunts {
    rekey_time = 0
    local_addrs = 127.0.0.1
    remote_addrs = 127.0.0.1
    children {
      pass-dhcp {
        local_ts  = 0.0.0.0/0
        remote_ts = 0.0.0.0/0[udp/68],0.0.0.0/0[udp/67]
        mode = pass
        start_action = trap
      }
      pass-ntp {
        local_ts  = 0.0.0.0/0
        remote_ts = 0.0.0.0/0[udp/123]
        mode = pass
        start_action = trap
      }
      pass-icmp {
        local_ts  = 0.0.0.0/0
        remote_ts = 0.0.0.0/0[icmp]
        mode = pass
        start_action = trap
      }
      drop-rwv4 {
        local_ts  = 0.0.0.0/0
        remote_ts = 0.0.0.0/0
        mode = drop
        start_action = trap
      }
      drop-rwv6 {
        local_ts  = ::/0
        remote_ts = ::/0
        mode = drop
        start_action = trap
      }
    }
  }
}
```
My question is: Is it good practice to use xfrm block and pass policies to secure an IPSEC roadwarrior when there are no tunnels ?

**Comment by tobiasbrunner** [comment, marked as answer]
> Is it good practice to use xfrm block and pass policies to secure an IPSEC roadwarrior when there are no tunnels ?

Sure, that's what they are there for. You could also use firewall rules (with policy match) with similar results. I guess it depends on your preferences which approach you take.
